### DISCUSSION TITLE
Why Reaction SMARTS and not SMIRKS?

### DISCUSSION THREAD

**Discussion by jasondbiggs** [rdkit/rdkit#5168] [Q&A]
In the [RDKit Book](https://www.rdkit.org/docs/RDKit_Book.html#reaction-smarts) it says

>Reaction SMARTS
>
>Not SMIRKS, not reaction SMILES , derived from SMARTS

What exactly is the distinction?  Are there examples of SMIRKS strings that can't be parsed by the RDKit?  I am thinking about this with respect to John Mayfield's [blog post](http://efficientbits.blogspot.com/2018/04/rdkit-reaction-smarts.html) on the topic.

**Comment by greglandrum** [comment]
There are too many words for closely related things, that's why I wrote that section of the RDKit documentation. It attempts to describe what we've implemented.

When I did the original implementation I was concerned with providing a way of expressing both reaction queries and transformations as flexibly and compactly as possible. Given how difficult the problem is, I was not concerned with being able to parse similar-looking formats which were developed for other toolkits. The name "Reaction SMARTS" is a bit unfortunate since that does actually show up in the Daylight SMARTS documentation.

I believe that the RDKit Reaction SMARTS parser should parse, and probably correctly interpret, any valid SMIRKS. However, given that SMIRKS often contain explicit H atoms, those reactions are unlikely to directly work with standard RDKit molecules. We could add a SMIRKS parser (really just a post-processing step for the reaction SMARTS parser) with relatively little work, but without access to a large library of actual SMIRKS from a reliable source, I would be extremely reluctant to undertake this.


> Aside: When someone first pointed out that "Reaction SMARTS" is actually a Daylight thing I was quite surprised since I'm normally more careful than that. However "Reaction SMARTS" seems to have appeared in the Daylight docs in the summer of 2006, which is right around when I started the RDKit reaction handling code... I must have been working with an older version of the SMARTS/SMIRKS docs at the time.
